nyaa/search.py

```diff
@@ -3,15 +3,99 @@
 import math
 import json
 import shlex
+from datetime import datetime, timedelta
 
 from nyaa import app, db
-from nyaa import models
+from nyaa import models, torrents
 
 import sqlalchemy_fulltext.modes as FullTextMode
 from sqlalchemy_fulltext import FullTextSearch
 from elasticsearch import Elasticsearch
 from elasticsearch_dsl import Search, Q
 
+class ElasticTorrentStatsAccessor:    
+    def __init__(self, elasticResult):
+        self.seed_count = elasticResult.seed_count
+        self.leech_count = elasticResult.leech_count
+        self.download_count = elasticResult.download_count
+
+class ElasticTorrentAccessor:
+    knownAttributes = ['id', 'has_torrent', 'display_name', 'filesize', 'anonymous', 'trusted', 'remake', 'complete', 'hidden', 'deleted']
+
+    def __init__(self, elasticResult):
+        self.elastic = elasticResult
+
+    def __getattr__(self, name):
+        # Passthru for attributes that are the same
+        if name in self.knownAttributes:
+            return self.elastic[name]
+
+        # Safetynet: throw if an attribute is accessed which we don't implement    
+        raise Exception("Unknown attribute {}".format(name))
```

Comment:
I wanted to cover the case when someone in the future adds something to the db, like created_time and adds them to the template, but forgets to handle it in the wrapper. created_time exists in both elastic and db, it's just a different type, so without the check it would fail silently and render an incompatible type in the ui.
But that hinges on someone else doing something wrong in the future. Your proposal is definitely lower maintenance.